localization/ndt_scan_matcher/src/map_update_module.cpp

```diff
@@ -45,6 +49,57 @@ MapUpdateModule::MapUpdateModule(
   need_rebuild_ = true;
 }
 
+void MapUpdateModule::initialize_diagnostics_key_value()
+{
+  diagnostics_map_update_->addKeyValue("timer_callback_time_stamp", 0.0);
+  diagnostics_map_update_->addKeyValue("is_activated", false);
+  diagnostics_map_update_->addKeyValue("is_set_last_update_position", false);
+  diagnostics_map_update_->addKeyValue("distance_last_update_position_to_current_position", 0.0);
+  diagnostics_map_update_->addKeyValue("is_need_rebuild", false);
+  diagnostics_map_update_->addKeyValue("is_updated_map", false);
+  diagnostics_map_update_->addKeyValue("is_succeed_call_pcd_loader", false);
+  diagnostics_map_update_->addKeyValue("maps_to_add_size", 0);
+  diagnostics_map_update_->addKeyValue("maps_to_remove_size", 0);
+  diagnostics_map_update_->addKeyValue("maps_size", 0);
+  diagnostics_map_update_->addKeyValue("latest_update_execution_time", 0.0);
+}
+
+void MapUpdateModule::callback_timer(
+  const bool is_activated, const std::optional<geometry_msgs::msg::Point> & position)
+{
+  diagnostics_map_update_->addKeyValue("timer_callback_time_stamp", clock_->now().seconds());
+  diagnostics_map_update_->addKeyValue("is_activated", is_activated);
+
+  if (is_activated) {
+    const bool is_set_last_update_position = (position != std::nullopt);
+    diagnostics_map_update_->addKeyValue(
+      "is_set_last_update_position", is_set_last_update_position);
+
+    if (is_set_last_update_position) {
+      if (should_update_map(position.value())) {
+        RCLCPP_INFO(logger_, "Start updating NDT map (timer_callback)");
+        update_map(position.value());
+      }
```

Comment:
I think it would be better to incorporate `should_update_map` into `update_map` and remove the if statement here.

Reply:
`update_map` is also used in `ndt_align_service`, so I don't think it can be simply incorporated.